src/utils/stacktrace.ts

```diff
@@ -0,0 +1,95 @@
+import ErrorStackParser from "error-stack-parser";
+
+export type RawStackFrames = string;
+
+const getErrorTitle = (e: Error): string => `${e.name || "Error"}: ${e.message}`;
+const getErrorStackFrames = (e: Error): string | undefined => e.stack?.replace(getErrorTitle(e) + "\n", "");
+
+export function captureRawStackFrames(filterFunc?: (...args: unknown[]) => unknown): RawStackFrames {
+    const savedStackTraceLimit = Error.stackTraceLimit;
+    const targetObj = {} as { stack: RawStackFrames };
+
+    Error.stackTraceLimit = 64;
+    Error.captureStackTrace(targetObj, filterFunc || captureRawStackFrames);
+    Error.stackTraceLimit = savedStackTraceLimit;
+
+    const rawFramesPosition = targetObj.stack.indexOf("\n") + 1; // crop out error message
+
+    return targetObj.stack.slice(rawFramesPosition);
+}
+
+export function applyStackFrames(error: Error, frames: RawStackFrames): Error {
+    if (!error || !error.message) {
+        return error;
+    }
+
+    error.stack = getErrorTitle(error) + "\n" + frames;
+
+    return error;
+}
+
+export function filterExtraWdioFrames(error: Error): Error {
+    if (!error || !error.message || !error.stack) {
+        return error;
+    }
+
+    const frames = getErrorStackFrames(error)!;
+    const rawFramesArr = frames.split("\n");
+    const framesParsed = ErrorStackParser.parse(error);
+
+    if (rawFramesArr.length !== framesParsed.length) {
+        return error;
+    }
```

Comment:
We assume "each line of rawFramesArr is equal to framesParsed"
Because we filter "rawFramesArr" using "framesParsed".
I dont know if `rawFramesArr.length !== framesParsed.length` even could be true, but just in case